macro/src/lib.rs

```diff
@@ -317,134 +317,395 @@ impl FieldParser {
+                unsafe {
+                    self.#steal_func_name_unchecked(index)
+                }
+            }
 
-                    #[doc = "Obtain a shared instace of the inner MMIO field `"]
-                    #[doc = stringify!(#field_ident)]
-                    #[doc = "`."]
-                    #[doc = "This variant only allow access to non-mutable methods of the MMIO block,"]
-                    #[doc = "but only requires a shared reference to the outer MMIO block."]
-                    #[doc = ""]
-                    #[doc = "# Lifetime and Safety"]
-                    #[doc = ""]
-                    #[doc = "The lifetime of the returned inner MMIO block is static which"]
-                    #[doc = "allows independent usage of the inner block and arbitrary"]
-                    #[doc = "creation of of multiple inner blocks for the same peripheral."]
-                    #[doc = "If you create multiple instances of this handle at the same time,"]
-                    #[doc = "you are responsible for ensuring that there are no read-modify-write"]
-                    #[doc = "races on any of the registers."]
-                    #[inline]
-                    pub fn #field_ident_shared(&self) -> derive_mmio::SharedInner<#inner_mmio_path<'_>> {
-                        derive_mmio::SharedInner::__new_internal(
-                            unsafe {
-                                self.#steal_func_unchecked_name()
-                            }
-                        )
+            #[doc = "Obtain a shared instace of the inner MMIO field `"]
+            #[doc = stringify!(#field_ident)]
+            #[doc = "`."]
+            #[doc = "This variant only allow access to non-mutable methods of the MMIO block,"]
+            #[doc = "but only requires a shared reference to the outer MMIO block."]
+            #[doc = ""]
+            #[doc = "# Lifetime and Safety"]
+            #[doc = ""]
+            #[doc = "The lifetime of the returned inner MMIO block is static which"]
+            #[doc = "allows independent usage of the inner block and arbitrary"]
+            #[doc = "creation of of multiple inner blocks for the same peripheral."]
+            #[doc = "If you create multiple instances of this handle at the same time,"]
+            #[doc = "you are responsible for ensuring that there are no read-modify-write"]
+            #[doc = "races on any of the registers."]
+            #[inline]
+            pub fn #field_ident_shared(&self, index: usize) -> Result<derive_mmio::SharedInner<#inner_mmio_path<'_>>, #error_type> {
+                if index >= self.#array_len_func() {
+                    return Err(#error_type(index));
+                }
+                // Safety: Index was checked.
+                Ok(unsafe { self.#field_ident_shared_unchecked(index) })
+            }
+
+            #[doc = "Obtain a shared instace of the inner MMIO field `"]
```

Comment:
```suggestion
            #[doc = "Obtain a shared instance of the inner MMIO field `"]
```

Reply:
I really am a typo specialist. I probably should try out https://github.com/crate-ci/typos :D